test/lib/extractor/oci-distribution-metadata.spec.ts

```diff
@@ -0,0 +1,233 @@
+        manifestDigest:
+          "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+        platform: "linux/amd64",
+      },
+      undefined,
+    ],
+    [
+      "given a platform that is too long should return undefined",
+      {
+        imageName: "gcr.io/example/repo:test",
+        manifestDigest:
+          "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+        // 64 is the maximum length, there are 65 characters here
+        platform:
+          "linux/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+      },
+      undefined,
+    ],
+    [
+      "given an image name with an invalid host should return undefined",
+      {
+        imageName: "..io/repo:test",
+        manifestDigest:
+          "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+        platform: "linux/amd64",
+      },
+      undefined,
+    ],
+    [
+      "given an image name with an invalid namespace should return undefined",
+      {
+        imageName: "gcr.io/re*&&po:test",
+        manifestDigest:
+          "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+        platform: "linux/amd64",
+      },
+      undefined,
+    ],
+    [
+      "given an image name with an invalid tag should return undefined",
+      {
+        imageName: "gcr.io/example/repo:__*image=",
+        manifestDigest:
+          "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+        platform: "linux/amd64",
+      },
+      undefined,
+    ],
+    [
+      "given an invalid manifest digest should return undefined",
+      {
+        imageName: "gcr.io/example/repo:test",
+        manifestDigest: "sha256:abc",
+        platform: "linux/amd64",
+      },
+      undefined,
+    ],
+    [
+      "given an invalid index digest should return undefined",
+      {
+        imageName: "gcr.io/example/repo:test",
+        indexDigest: "sha256:abc",
+        manifestDigest:
+          "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+        platform: "linux/amd64",
+      },
+      undefined,
+    ],
+    [
+      "given an invalid platform should return undefined",
```

Comment:
should we add some logging to detail why undefined is returned seeing as there are a number of cases for this result?

Reply:
Declining to add logging here because this is a library and we don't have a good mechanism to support BYOL.